### ISSUE TITLE
Support multiple attachments

### ISSUE THREAD

**Issue by driskull** [Esri/geoform-template-js#217]
Users may want to attach multiple files?

**Comment by CTDisasterManagement** [comment]
@driskull 
In case the attachment is configured to be a required field in the form, Is it o.k. that the attachment upload fails and at the same time the apply edits operation succeeds?
Thoughts!

Thanks!

**Comment by driskull** [comment]
Hmm. Doesnt the applyEdits happen first? I think it has to because we need the ID of the record to upload, correct?

I don't think we can force them to upload once submitted so I think thats ok.

**Comment by CTDisasterManagement** [comment]
Yes, the applyEdits happens first. And only then we can upload the file.
We just wanted you to do a quick check with you on this scenario.
Thank you for your confirmation @driskull !

**Comment by CTDisasterManagement** [comment]
@driskull 
Attaching screens of file-upload modal window:

* Screen-shots for 'success' and 'uploading' status:
![image](https://cloud.githubusercontent.com/assets/5261405/5919224/cdf6c7b2-a657-11e4-912e-913a513a29bd.png)

* Screen-shot  for 'retry' status:
![image](https://cloud.githubusercontent.com/assets/5261405/5919607/7912d9b2-a65b-11e4-8987-7e339f495de1.png)

Thoughts!!

Thanks!

**Comment by driskull** [comment]
The retry should look more like a button. Currently, it just looks like a status. Is it using a button class?

**Comment by julio8a** [comment]
let's use these icons from bootstrap for displaying completed and retry...

![image](https://cloud.githubusercontent.com/assets/6144462/5922826/85e9dee0-a603-11e4-923e-6365ff7acb1f.png)

**Comment by driskull** [comment]
Instead of "Retry..." lets say "Error".
Instead of "Success..." lets say "Uploaded"
Have a button next to "Error" that says "Retry".

**Comment by CTDisasterManagement** [comment]
@driskull 
Yes, the retry button is using button class.The retry text is not looking like a button because in that specific theme the background color for alert-danger and btn-danger is same. The effect of button for this theme comes only after hover as shown in the first failed entry in screenshot above.

Also attaching a screenshot with a different theme where the alert-danger and btn-danger classes have different background colors and the button is distinguishable.
![image](https://cloud.githubusercontent.com/assets/5261405/5954316/d3b8372a-a7bb-11e4-92bb-f0002afe968f.png)

**Comment by CTDisasterManagement** [comment]
@driskull 
Do we need to include the glyphicons or just the text suggested?

**Comment by driskull** [comment]
I think the glyph icons would be helpful.

Let's just make sure we have the two things separated:

1. A status area that shows uploading, success, error. (not a button)
2. If necessary, a button to retry.

**Comment by CTDisasterManagement** [comment]
@driskull 
After string freeze we found out a bug related to select file button title issue and to fix it we need to add a string in nls
Please let us know if we can send this change across
Thanks!

**Comment by driskull** [comment]
Yes please add it now

- Matt


On Feb 2, 2015, at 5:52 AM, CTDisasterManagement <notifications@github.com<mailto:notifications@github.com>> wrote:


@driskull<https://github.com/driskull>
After string freeze we found out a bug related to select file button title issue and to fix it we need to add a string in nls
Please let us know if we can send this change across
Thanks!

**Comment by CTDisasterManagement** [comment]
@driskull 
New pull request with above mentioned nls changes sent.
Thanks!

**Comment by CTDisasterManagement** [comment]
@driskull 
The add attachment functionality is not working with IE8 and IE9.
We get 'access is denied' error.
And also the file size cannot be determined in IE8 and IE9 as it is an HTML5 capability.
Thanks!

**Comment by driskull** [comment]
Are we modifying the file input via javascript?

IE doesn't allow manipulation of the type="file" input element from javascript due to security reasons. Setting the filename or invoking a click event to show the browser dialog will result in an "Access is denied" error on the form submit - Internet Explorer is clever about remembering what methods have been invoked.

**Comment by CTDisasterManagement** [comment]
@driskull 
No, we are not manipulating the file input element.
We are just picking the file input as it is and putting it into a form element.
And all such forms are pushed into an array, and then we are using this array to add attachment.
Manipulation in file input would not let the attachment happen in any browser, whereas this issue is ie8 and ie9 specific.
Thanks!

**Comment by driskull** [comment]
@CTDisasterManagement is it happening on click? http://stackoverflow.com/questions/7258226/jquery-input-file-click-method-and-access-denied-on-ie

**Comment by CTDisasterManagement** [comment]
@driskull 
We made a POC for file attachment having a simple file input and it worked in IE8.

Need to work  on it to implement the same with multiple attachments in our application.

Will keep you posted on the updates.

Thanks!

**Comment by driskull** [comment]
Thanks!

**Comment by CTDisasterManagement** [comment]
Hi @driskull ,

The multiple attachments is now working in IE8 and IE9 as well.

Here we would like to mention that the HTML5 File API is not supported in IE8 and IE9 and hence the size of file cannot be determined in these versions of IE.

Hence the two functionalities of showing the size of selected file and throwing an error on exceeding maximum file size limit of 25 MB is not supported in IE8 & 9.

We will send out a pull request for the above once our QA team completes a round of testing.

Thanks!

**Comment by driskull** [comment]
Ok great.
